Initial commit of far-field voice HAL interface for review

garys2511 opened the pull request.

garys2511 committed:
Initial commit of far-field voice HAL interface for review

dwolaver commented on src/xr-audio/ffv_hal_interface/FarFieldVoiceController.h:
```diff
@@ -0,0 +1,381 @@
+     * 
+     * @see closeKeywordChannel()
+     */
+	::hal::BinderStatus openKeywordChannel(::hal::HalFileDescriptor** fileDescriptor);
```
suggest openChannel or similar.  Optional parameters can be passed to indicate that the channel shall perform keyword detection and any required callbacks etc.

dwolaver commented on src/xr-audio/ffv_hal_interface/FarFieldVoiceController.h:
```diff
@@ -0,0 +1,381 @@
+     * 
+     * @see closeContinualChannel()
+     */
+	::hal::BinderStatus openContinualChannel(::hal::HalFileDescriptor** fileDescriptor);
```
all the "Continual" apis are redundant and should be removed.

dwolaver commented on src/xr-audio/ffv_hal_interface/FarFieldVoiceController.h:
```diff
@@ -0,0 +1,381 @@
+     * 
+     * @see closeMicrophoneChannels()
+     */
+	::hal::BinderStatus openMicrophoneChannels(::hal::HalFileDescriptor** fileDescriptor);
```
The "Microphone" channels api's also seem redundant and should be an optional parameter to the openChannel api.

dwolaver reviewed (changes requested):
The voice SDK is written entirely in C.  The twelve header files should be consolidated into a single (or a few) header files.  It should follow the existing style and format of the header files in the voice sdk (ie consistency).

garys2511 committed:
Update scope resolution to qualified names. Add compile switch for Win32 builds.

garys2511 committed:
RDKEMW-5977 : FFV HAL Interface

Reason for change: Update to C/C++ wrapper header to sync with AIDL changes

Test Procedure: N/A

Priority: P2

Risks: None

Signed-off-by: Gary_Skrabutenas@comcast.com

garys2511 committed:
RDKEMW-5977 : FFV HAL Interface

Reason for change: Updated multiple header files, added internal class
FFVmsgQueue, changed name KeywordDetectData.h to KeywordDetectInfo.h

Test Procedure: N/A

Priority: P2

Risks: None

Signed-off-by: Gary_Skrabutenas@comcast.com

garys2511 committed:
RDKEMW-5977 : FFV HAL Interface

Reason for change: Update to C wrapper header based on Aug 4 code review
feedback. Added FFVplatformDepended.h file

Test Procedure: N/A

Priority: P2

Risks: None

Signed-off-by: Gary_Skrabutenas@comcast.com
